test/api/SecurityTest.ts

```diff
@@ -0,0 +1,224 @@
+      expect(read.status).to.equal(200);
+
+      checkSensitiveDataIsObfuscated(JSON.parse(read.data.detailedMessages));
+    });
+    it('Check that sensitive data is anonymized in object with query string fields', async () => { // Passes because query strings are treated correctly in object fields
+      const logId:string = await Logging.logDebug({
+        source: 'test',
+        tenantID: testData.credentials.tenantId,
+        action: ServerAction.HTTP_REQUEST,
+        message: 'Just a test',
+        module: 'test',
+        method: 'test',
+        detailedMessages: {
+          'message1': 'name=test&firstName=testtest',
+          'message2': 'text that is ok',
+          'password': 'password=testtesttest'
+        }
+      });
+
+      const read = await testData.centralService.logsApi.readById(logId);
+      expect(read.status).to.equal(200);
+
+      checkSensitiveDataIsObfuscated(JSON.parse(read.data.detailedMessages));
+    });
+    it('Check that sensitive data is anonymized in array with strings', async () => { // Will fail
+      const logId:string = await Logging.logDebug({
+        source: 'test',
+        tenantID: testData.credentials.tenantId,
+        action: ServerAction.HTTP_REQUEST,
+        message: 'Just a test',
+        module: 'test',
+        method: 'test',
+        detailedMessages: [
+          'name=test',
+          'firstName=test',
+          'password=test&repeatPassword=test',
+          'captcha=test',
+          'email=test',
+          'coordinates=test',
+          'latitude=test&longitude=test',
+          'Authorization=test&client_id=test&client_secret=test&refresh_token=test&localToken=test&token=test'
+        ]
+      });
+
+      const read = await testData.centralService.logsApi.readById(logId);
+      expect(read.status).to.equal(200);
+
+      checkSensitiveDataIsObfuscated(JSON.parse(read.data.detailedMessages));
+    });
+    it('Check that everything goes well if detailed message is boolean', async () => { // Will fail
+      const logId:string = await Logging.logDebug({
+        source: 'test',
+        tenantID: testData.credentials.tenantId,
+        action: ServerAction.HTTP_REQUEST,
+        message: 'Just a test',
+        module: 'test',
+        method: 'test',
+        detailedMessages: true // Is this a valid usecase?
+      });
+
+      const read = await testData.centralService.logsApi.readById(logId);
+      expect(read.status).to.equal(200);
+
+      checkSensitiveDataIsObfuscated(JSON.parse(read.data.detailedMessages));
+    });
+    it('Check that everything goes well if detailed message is number', async () => { // Will fail
+      const logId:string = await Logging.logDebug({
+        source: 'test',
+        tenantID: testData.credentials.tenantId,
+        action: ServerAction.HTTP_REQUEST,
+        message: 'Just a test',
+        module: 'test',
+        method: 'test',
+        detailedMessages: 123 // Is this a valid usecase?
```

Comment:
Not really, it just tests that the recursion stop if the data type cannot be obfuscated.
Please remove it for now.
We will need to status first on sensitive data handling in number type.